### PR TITLE
coll: prompt large count in allgatherv and iallgatherv algorithms

### PR DESCRIPTION
**Pull request by zhenggb72**
## Pull Request Description

change local variables used in allgatherv algorithms such as count and offset datatype from int to MPI_Aint to work with large count.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

File: src/mpi/coll/iallgatherv/iallgatherv_tsp_recexch.c [hunk 1]
```diff
@@ -18,8 +18,8 @@ static int MPIR_TSP_Iallgatherv_sched_intra_recexch_data_exchange(int rank, int
 {
     int mpi_errno = MPI_SUCCESS;
     int mpi_errno_ret ATTRIBUTE((unused)) = MPI_SUCCESS;
-    int partner, offset, count;
-    int i, vtx_id;
+    int partner;
+    int count, offset, i, vtx_id;
```
Comment (hzhou) [inline src/mpi/coll/iallgatherv/iallgatherv_tsp_recexch.c:22]: Did you mean to use `MPI_Aint`?

File: src/mpi/coll/iallgatherv/iallgatherv_tsp_recexch.c [hunk 2]
```diff
@@ -122,8 +122,9 @@ int MPIR_TSP_Iallgatherv_sched_intra_recexch_step2(int step1_sendto, int step2_n
 {
     int mpi_errno = MPI_SUCCESS;
     int mpi_errno_ret ATTRIBUTE((unused)) = MPI_SUCCESS;
-    int phase, i, j, count, nbr, offset, rank_for_offset;
-    int x, vtx_id;
+    int phase, i, j, nbr, rank_for_offset;
+    MPI_Aint x;
+    int count, offset, vtx_id;
```
Comment (hzhou) [inline src/mpi/coll/iallgatherv/iallgatherv_tsp_recexch.c:127]: Does `count, offset` need be in `MPI_Aint`?

File: src/mpi/coll/iallgatherv/iallgatherv_tsp_recexch.c (hunk 1 above)
Comment (zhenggb72) [reply src/mpi/coll/iallgatherv/iallgatherv_tsp_recexch.c]: count offset are calculated by MPII_Recexchalgo_get_count_and_offset, which returns the rank number and rank offset. I think int is fine.

File: src/mpi/coll/iallgatherv/iallgatherv_tsp_recexch.c (hunk 2 above)
Comment (zhenggb72) [reply src/mpi/coll/iallgatherv/iallgatherv_tsp_recexch.c]: same

**Comment by hzhou** [comment]
test:mpich/ch3/most
test:mpich/ch4/most

**Comment by hzhou** [comment]
@zhenggb72 Please add PR description and commit messages.

**Comment by zhenggb72** [comment]
> @zhenggb72 Please add PR description and commit messages.

added

**Comment by hzhou** [comment]
test:mpich/ch3/most
test:mpich/ch4/most

**Review by hzhou — APPROVED** [review]
LGTM

**Pull request merged**